test/testother.cpp

```diff
@@ -3371,6 +3374,15 @@ class TestOther : public TestFixture {
         ASSERT_EQUALS("[test.cpp:1]: (style) Parameter 's1' can be declared as reference to const\n"
                       "[test.cpp:1]: (style) Parameter 's2' can be declared as reference to const\n",
                       errout.str());
+
+        check("struct S {\n"
+              "    void f(int& r) { p = &r; }\n"
+              "    int* p;\n"
+              "};\n"
+              "void g(std::vector<int>& v1, std::vector<int*>& v2) {\n"
+              "    std::transform(v1.begin(), v1.end(), v2.begin(), [](auto& x) { return &x; });\n"
```

Comment:
This can't be handled by `isVariableChanged`. We need to check if it escapes. Ideally it would be nice to write a generic function(maybe `escapes` or `referenceEscapes`?) to check for this(with an `indirect` parameter for pointers and references), this should help to simplify the logic for this checker.